nano/node/node.cpp

```diff
@@ -3413,15 +3542,60 @@ void nano::active_transactions::request_confirm (std::unique_lock<std::mutex> &
 				}
 				if ((!reps->empty () && total_weight > node.config.online_weight_minimum.number ()) || roots_size > 5)
 				{
-					if (confirm_req_bundle.size () < max_broadcast_queue)
+					// broadcast_confirm_req_base modifies reps, so we clone it once to avoid aliasing
+					if (nano::nano_network != nano::nano_networks::nano_test_network)
 					{
-						confirm_req_bundle.push_back (std::make_pair (i->election->status.winner, reps));
+						if (confirm_req_bundle.size () < max_broadcast_queue)
+						{
+							confirm_req_bundle.push_back (std::make_pair (i->election->status.winner, reps));
+						}
+					}
+					else
+					{
+						for (auto j (reps->begin ()), m (reps->end ()); j != m; j++)
+						{
+							auto rep_request (requests_bundle.find (j->endpoint));
+							auto block (i->election->status.winner);
+							auto root_hash (std::make_pair (block->hash (), nano::uint512_union (block->previous (), block->root ())));
+							if (rep_request == requests_bundle.end ())
+							{
+								std::vector<std::pair<nano::block_hash, nano::uint512_union>> insert_vector (1, root_hash);
+								if (requests_bundle.size () < max_broadcast_queue * 2)
+								{
+									requests_bundle.insert (std::make_pair (j->endpoint, insert_vector));
+								}
+							}
+							else if (rep_request->second.size () < max_broadcast_queue * 6)
```

Comment:
How is this magic number chosen? (and the one above it). If it's dependent on something can it be derived from that? or use a variable describing it or a comment.

Reply:
6 is max hashes in confirm_req (now const in network), while above should be just max_broadcast_queue